### PR TITLE
Allan/skill question update

### PR DESCRIPTION
**Pull request by AllanT102**
🎟️ Ticket(s): Closes #

👷 Changes: Created new skills question for partners to fill out and removed skills checkbox. Also changed partner registration form to have selection dropdown for pronouns.

💭 Notes: Any additional things to take into consideration.

Wait! Before you merge, have you checked the following:

📷 Screenshots
Screen Shot 2023-09-17 at 9 18 24 PM

Screen Shot 2023-09-17 at 9 17 50 PM

(prefer animated gif)

## Checklist

- [ ] Looks good on large screens
- [ ] Looks good on mobile

### CODE REVIEW

**Comment by jerryxu99** [comment]
its missing the options part it seems

also pics next time pls

+ let's remove the skills question type for regular Attendees form and keep it only specific to the partner registration form

**Comment by AllanT102** [comment]
@jerryxu99 What options part?

Also, do we want to remove this new skills question from the attendee form customer questions? I think it would be ok to have it as an option as it might be useful to filter applicants for competitions? idk thoughts?

**Comment by jerryxu99** [comment]
the options part as in now its a text box, whereas before it was a selection. In visuals it should look exactly the same as the Checkbox question type as it did before

Imo its pretty useless for aplicants to self diagnose skills for themselves for competitions so id say remove

**Comment by AllanT102** [comment]
Is it necessary to have checkboxes then? There's no way to list every skill out there, so wouldn't it be better to just put an example of skills that the mentor can input? so the skills question would be: "Which of the following skills/descriptors apply to you ("Java, Python, Product Management, BUCS Alumni, MIS Grad")?"

**Comment by jerryxu99** [comment]
omg u did the pronoun change tyvm

i think the ideal is to have it be like what it was for MIS night, where the event director is able to define a couple skills and include an _other_ box for any others. It's important that the event director creates a list of basic ones to choose from so that we have consistency when people search for skills on the companion app.

**Comment by jerryxu99** [comment]
also @ddennis924 lets be careful when we roll out this feature bc the hello hacks registrant form might be live when this gets pushed. We'd have to manually edit the partner form to add back the skills question

@AllanT102 this change won't change how the mentors component detects the skills question will it?

**Comment by AllanT102** [comment]
Ok ill add the checkbox options + input.  this wont affect how mentors component detects the skills question

**Comment by AllanT102** [comment]
Skills quesiton now uses the checkboxes. Companion won't work right now because the way the skills are stored is now a list, whereas before it was a string. However, in the future, no change will be needed because the mentors page is changed to parse the array of skills instead of the string

File: src/pages/public/Companion/components/mentor/Mentors.js
Comment (jerryxu99) [inline src/pages/public/Companion/components/mentor/Mentors.js:120]: lets keep it returning a skill.trim() here so that we don't have to worry abt uncleaned data when working with dynamicResponses[skillQuestionId] in other areas

**Review by jerryxu99 — CHANGES_REQUESTED** [review]
one more thing from me sorry (see comment)

other than that lgtm, but let's wait until after Hellohacks finishes to merge this as the hellohacks reg form has alrdy been created

**Comment by jerryxu99** [comment]
what are the changes we need to make to companion app to link it all tgt? You mentioned companion app won't work after this right

**Comment by ddennis924** [comment]
> what are the changes we need to make to companion app to link it all tgt? You mentioned companion app won't work after this right

@AllanT102

**Review by ddennis924 — APPROVED** [review]

**Comment by AllanT102** [comment]
@ddennis924 oops sorry i keep forgetting to merge this change in. Can you approve again?

**Review by ddennis924 — APPROVED** [review]

**Pull request merged**
